barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover_internal.hpp

```diff
@@ -292,18 +296,28 @@ template <class DeciderProvingKeys_> class ProtogalaxyProverInternal {
      */
 
     template <size_t skip_count = 0>
-    static void extend_univariates(
-        std::conditional_t<skip_count != 0, ExtendedUnivariates, ExtendedUnivariatesNoOptimisticSkipping>&
+    BB_INLINE static void extend_univariates(
+        std::conditional_t<
+            Flavor::USE_SHORT_MONOMIALS,
+            ShortUnivariates,
+            std::conditional_t<skip_count != 0, ExtendedUnivariates, ExtendedUnivariatesNoOptimisticSkipping>>&
             extended_univariates,
         const DeciderPKs& keys,
         const size_t row_idx)
     {
         PROFILE_THIS_NAME("PG::extend_univariates");
-        auto incoming_univariates = keys.template row_to_univariates<ExtendedUnivariate::LENGTH, skip_count>(row_idx);
-        for (auto [extended_univariate, incoming_univariate] :
-             zip_view(extended_univariates.get_all(), incoming_univariates)) {
-            incoming_univariate.template self_extend_from<NUM_KEYS>();
-            extended_univariate = std::move(incoming_univariate);
+
+        if constexpr (Flavor::USE_SHORT_MONOMIALS) {
+            extended_univariates = std::move(keys.row_to_short_univariates(row_idx));
+        } else {
+            constexpr size_t _skip_count = Flavor::USE_SHORT_MONOMIALS ? 0 : skip_count;
```

Comment:
is this still necessary? given the folding flavor (Mega) is now using short monomials by default, doesn't that mean we're never skipping values in this function and the `skip_count` logic should be removed? Maybe I'm misunderstanding something?

Reply:
you are correct. I kept this code in in case we ever wanted to fold a flavor that was not amenable to this monomial abstraction. For example the AVM, ECCVM or Translator circuits. Wdyt should we keep this code path in or remove?

Reply:
I think I would personally incline for removing, we can always bring it back from git history, this code is a little monster already.